snappi_ixnetwork/trafficitem.py

```diff
@@ -191,9 +220,245 @@ class TrafficItem(CustomField):
+            self._api.assistant._ixnetwork.href
+        )
+        return (url, payload)
+
+    def _export_config(self):
+        href = "%sresourceManager" % self._api._ixnetwork.href
+        url = "%s/operations/exportconfig" % href
+        payload = {
+            "arg1": href,
+            "arg2": ["/traffic/trafficItem/descendant-or-self::*"],
+            "arg3": True,
+            "arg4": "json",
+        }
+        res = self._api._request("POST", url=url, payload=payload)
+        return json.loads(res["result"])
+
+    def _importconfig(self, imports):
+        imports["xpath"] = "/"
+        href = "%sresourceManager" % self._api._ixnetwork.href
+        url = "%s/operations/importconfig" % href
+        import json
+
+        payload = {
+            "arg1": href,
+            "arg2": json.dumps(imports),
+            "arg3": False,
+            # "arg4": "suppressNothing",
+            # "arg5": True,
+        }
+        try:
+            # TODO for larger config rest api is throwing error,
+            # with no url found, when the first response is 202 (in-progress)
+            # its keep checking the status of the url with 1 sec sleep, and
+            # after a while error is thrown. but could see the configuration
+            # applied at Ixnetwork. (Need to check with Eng team)
+            self._api._request("POST", url=url, payload=payload)
+        except Exception:
+            return
+        return
+
+    def get_ports_xpath(self, config):
+        ixn = self._api.assistant._ixnetwork
+        myfilter = [{"property": "name", "regex": ".*"}]
+        url, payload = self._get_search_payload(
+            "/",
+            "(?i)^(vport)$",
+            ["name"],
+            myfilter,
+        )
+        result = ixn._connection._execute(url, payload)
+        vports = {}
+        for vp in result:
+            if vp.get("vport") is None:
+                continue
+            for v in vp["vport"]:
+                vports[v["name"]] = v["xpath"]
+        return vports
+
+    def get_device_xpath(self, config):
+        if len(config.devices) == 0:
+            return {}
+        dev_names = []
+        for f in config.flows:
+            if f.tx_rx.choice == "port":
+                continue
+            dev_names.extend(f.tx_rx.device.tx_names)
+            dev_names.extend(f.tx_rx.device.rx_names)
+        dev_names = list(set(dev_names))
+        if dev_names == []:
+            return {}
+        selects = []
+        url = None
+        for name in dev_names:
+            href = self._api.get_ixn_href(name)
+            if href is None:
+                # TODO change to more detailed message
+                raise Exception(
+                    "href is not found for device name {}".format(name)
+                )
+            href = href.split("ixnetwork")[-1]
+            url, search = self._get_search_payload(
+                href, "^(xpath|ethernet|ipv4|ipv6|bgpv4|bgpv6)$", ["name"], []
```

Comment:
I think we need only xpath which select from device href. So is there any specific reason to provide this filter "^(xpath|ethernet|ipv4|ipv6|bgpv4|bgpv6)$"

Reply:
This is used to determine the traffic end point type, just with device href we cannot determine the traffictype, if we have these in place we can determine if ipv4, ipv6 and ethernetvlan.
But currently, this pattern is executed for all the hrefs. this can be modified to more specific. will update the code accordingly.
Thanks for mentioning